index.js

```diff
@@ -104,7 +104,37 @@ var authS3O = function (req, res, next) {
 	}
 };
 
+// Alternative authentication middleware which does not redirect to S3O when
+// cookies are missing or invalid. This can be used in front of API calls
+// where a redirect will be undesirable
+var authS3ONoRedirect = function (req, res, next) {
+	debug('S3O: Start.');
+
+	if (req.cookies === undefined || req.cookies === null) {
+		var cookies = req.headers.cookie;
+		if (cookies) {
+			req.cookies = cookieParser(cookies);
+		} else {
+			req.cookies = Object.create(null);
+		}
+	}
+
+	if (req.cookies.s3o_username && req.cookies.s3o_token && authenticateToken(res, req.cookies.s3o_username, req.hostname, req.cookies.s3o_token)) {
+		debug('S3O: Authentication succeeded');
+		return next();
+	};
```

Comment:
Most of the above logic is already done in `authS3O`.  Can we handle all that in a single place to avoid duplication and then just vary the response as appropriate between redirect and 403?